File: src/main/java/com/github/unchama/contextualexecutor/executors/PrintUsageExecutor.kt
package com.github.unchama.contextualexecutor.executors

import com.github.unchama.contextualexecutor.ContextualExecutor
import com.github.unchama.contextualexecutor.RawCommandContext
import com.github.unchama.messaging.asResponseToSender
import org.bukkit.command.Command

/**
 * コマンドの[Command.getUsage]を送信者に送り返すだけのアクションを返すExecutor
 */
object PrintUsageExecutor : ContextualExecutor {
  override suspend fun executeWith(rawContext: RawCommandContext) = rawContext.command.command.usage.asResponseToSender()
}

File: src/main/java/com/github/unchama/contextualexecutor/executors/PrintUsageExecutor.scala
package com.github.unchama.contextualexecutor.executors

import com.github.unchama.contextualexecutor.{ContextualExecutor, RawCommandContext}
import com.github.unchama.targetedeffect.MessageEffects._
import kotlin.coroutines.Continuation

/**
 * コマンドの[Command.usage]を送信者に送り返すだけのアクションを返すExecutor
 */
object PrintUsageExecutor extends ContextualExecutor {
  override def executeWith(rawContext: RawCommandContext, cont: Continuation[Unit]) =
      rawContext.command.command.getUsage.asMessageEffect().runFor(rawContext.sender, cont)
}
